tests/tools/test_edit.py

```
@@ -162,3 +162,146 @@ async def test_view_file_with_range_on_empty_file(edit_tool: EditTool, tmp_path:
+
+    tool = EditTool()
+
+    # Regex with LF only should still match CRLF due to tolerance
+    result = await tool(
+        command="str_replace",
+        path=str(test_file),
+        old_str=r"line2:\s+beta\nline3",
+        new_str="line2: gamma\nline3",
+        match_mode="regex",
+    )
+    assert result.error is None, f"Regex with CRLF tolerance failed: {result.error}"
```

Ruff (0.12.2)

Use of `assert` detected

(S101)